src/overlays/actors/ovl_En_Hanabi/z_en_hanabi.c

```diff
@@ -32,38 +33,328 @@ const ActorInit En_Hanabi_InitVars = {
+        if (arg0->unk_00 == 1) {
+            count++;
+        }
+    }
+
+    return count;
+}
+
+void func_80B22FA8(EnHanabiStruct* arg0, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    s32 i;
+    u8 sp53;
+
+    OPEN_DISPS(gfxCtx);
+
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    POLY_XLU_DISP = Gfx_CallSetupDL(POLY_XLU_DISP, 20);
+
+    gSPSegment(POLY_XLU_DISP++, 0x08, Lib_SegmentedToVirtual(gSun1Tex));
+    gSPDisplayList(POLY_XLU_DISP++, gameplay_keep_DL_07AB10);
+
+    sp53 = 0xFF;
+    if (sp53) {}
+
+    for (i = 0; i < 400; i++, arg0++) {
+        if (arg0->unk_00 == 1) {
+            Matrix_InsertTranslation(arg0->unk_08.x, arg0->unk_08.y, arg0->unk_08.z, MTXMODE_NEW);
+            Matrix_NormalizeXYZ(&globalCtx->billboardMtxF);
+            if (arg0->unk_01 < 40) {
+                Matrix_Scale(arg0->unk_04 * 0.025f * arg0->unk_01, arg0->unk_04 * 0.025f * arg0->unk_01, 1.0f,
+                             MTXMODE_APPLY);
+            } else {
+                Matrix_Scale(arg0->unk_04, arg0->unk_04, 1.0f, MTXMODE_APPLY);
+            }
+            Matrix_InsertZRotation_s(globalCtx->gameplayFrames * 4864, MTXMODE_APPLY);
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            if (sp53 != arg0->unk_02) {
+                gDPPipeSync(POLY_XLU_DISP++);
+                gDPSetEnvColor(POLY_XLU_DISP++, D_80B23C40[arg0->unk_02], D_80B23C40[arg0->unk_02 + 1],
+                               D_80B23C40[arg0->unk_02 + 2], 255);
+
+                sp53 = arg0->unk_02;
+            }
+
+            if (arg0->unk_01 < 6) {
+                gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, D_80B23C2C[arg0->unk_02], D_80B23C2C[arg0->unk_02 + 1],
+                                D_80B23C2C[arg0->unk_02 + 2], arg0->unk_01 * 50);
+            } else {
+                gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, D_80B23C2C[arg0->unk_02], D_80B23C2C[arg0->unk_02 + 1],
+                                D_80B23C2C[arg0->unk_02 + 2], 255);
+            }
+
+            gSPDisplayList(POLY_XLU_DISP++, gameplay_keep_DL_07AB58);
+        }
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void EnHanabi_Init(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnHanabi* this = THIS;
+    s32 i;
+
+    if (1) {}
+
+    Actor_SetScale(thisx, 1.0f);
+    this->actionFunc = func_80B23910;
+    this->unk_144 = 0;
+    func_80B22C00(this->unk_148);
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_4608); i++) {
+        Lights_PointNoGlowSetInfo(&this->unk_4608[i], thisx->world.pos.x, thisx->world.pos.y, thisx->world.pos.z, 255,
+                                  255, 255, -1);
+        this->unk_4634[i] = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->unk_4608[i]);
+    }
+
+    if ((thisx->params & 0x1F) == 1) {
```

Comment:
params access, should have a macro

Reply:
I was told before not to when it's only used in 1 place.